src/module/actor/prep/functions/MatrixPrep.ts

```diff
@@ -46,13 +47,14 @@ export class MatrixPrep {
             matrix.item = device.system;
             matrix.running_silent = device.isRunningSilent();
             const deviceAtts = device.getASDF();
-            if (deviceAtts) {
-                // setup the actual matrix attributes for the actor
-                for (const [key, value] of Object.entries(deviceAtts)) {
-                    if (value && matrix[key]) {
-                        matrix[key].base = value.value;
-                        matrix[key].device_att = value.device_att;
-                    }
+            // setup the actual matrix attributes for the actor
+            for (const [key, value] of Object.entries(deviceAtts)) {
+                if (value) {
```

Comment:
Not a change request, just a suggestion. 

Readability wise this would be improved by removing the nesting by using a guard clause:

```TypeScript
if (!value) continue;

// create a new attribute field from the current one, this also works if the matrix[key] field doesn't exist
const att = DataDefaults.createData('attribute_field', matrix[key]);
att.base = value.value;
att['device_att'] = value.device_att;
matrix[key] = att;
```